torchvision/prototype/datasets/utils/_resource.py

```diff
@@ -0,0 +1,54 @@
+import os.path
+import pathlib
+from typing import Optional, Union
+from urllib.parse import urlparse
+
+from torch.utils.data import IterDataPipe
+from torch.utils.data.datapipes.iter import FileLoader, IterableWrapper
+
+
+# FIXME
+def compute_sha256(_) -> str:
+    return ""
+
+
+class LocalResource:
+    def __init__(
+        self, path: Union[str, pathlib.Path], *, sha256: Optional[str] = None
+    ) -> None:
+        self.path = pathlib.Path(path).expanduser().resolve()
+        self.file_name = self.path.name
+        self.sha256 = sha256 or compute_sha256(self.path)
+
+    def to_datapipe(self) -> IterDataPipe:
+        return FileLoader(IterableWrapper((str(self.path),)))
+
+
+class OnlineResource:
+    def __init__(self, url: str, *, sha256: str, file_name: str) -> None:
+        self.url = url
+        self.sha256 = sha256
+        self.file_name = file_name
+
+    def to_datapipe(self, root: Union[str, pathlib.Path]) -> IterDataPipe:
+        path = (pathlib.Path(root) / self.file_name).expanduser().resolve()
+        # FIXME
+        return FileLoader(IterableWrapper((str(path),)))
```

Comment:
This is greatest weak point right now. Functionality to download stuff and verify the checksum is present in `torchdata`, but is not public yet.